input_test_cases/run_all_tests.py

```diff
@@ -0,0 +1,217 @@
+import os
+import subprocess
+
+import nibabel as nib
+import numpy as np
+import pandas as pd
+
+
+def test_impedance(input_dir: str, input_json: str, output_dir: str, desired_dir: str):
+    """
+    Tests impedance by comparing the actual output with the desired output.
+
+    Parameters
+    ----------
+    input_dir : str
+        Directory where the input files are located.
+    input_json : str
+        JSON file with input parameters.
+    output_dir : str
+        Directory where the output files are generated.
+    desired_dir : str
+        Directory where the desired output files are located.
+    """
+    os.chdir(input_dir)
```

Comment:
I am not a big fan of so many calls to `os.chdir`. 
I updated the structure of the code for one example to do it in an easier way.